tuning/trainercontroller/controllermetrics/metrics.py

```diff
@@ -0,0 +1,218 @@
+import numpy as np
+from transformers import TrainerState, IntervalStrategy
+from transformers.utils import logging
+from collections import deque
+import math
+import copy
+
+logger = logging.get_logger(__name__)
+
+class MetricHandler:
+    """Base class for the controller-metrics"""
+    def validate(self, training_args) -> bool:
+        pass
+    
+    def compute(self, training_state, training_args=None, metrics=None) -> dict:
+        pass   
+
+class WindowStepLoss(MetricHandler):
```

Comment:
Do we need Window in the name? We don't have it in Epoch loss.

Reply:
done